processor/ppoUndeliveredProcessor.go

```diff
@@ -0,0 +1,52 @@
+package processor
+
+import (
+	"context"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/config"
+	"github.com/ONSdigital/census-rm-pubsub-adapter/models"
+)
+
+type ppoUndeliveredProcessor struct {
+	*Processor
+}
+
+func NewPpoUndeliveredProcessor(ctx context.Context, appConfig *config.Configuration) *ppoUndeliveredProcessor {
+	ppoUndeliveredProcessor := &ppoUndeliveredProcessor{}
+	ppoUndeliveredProcessor.Processor = NewProcessor(ctx, appConfig, appConfig.PpoUndeliveredProject, appConfig.PpoUndeliveredSubscription, appConfig.UndeliveredRoutingKey, convertPpoUndeliveredToRmMessage, unmarshalPpoUndelivered)
+	return ppoUndeliveredProcessor
+}
+
+func unmarshalPpoUndelivered(data []byte) (models.PubSubMessage, error) {
+	var ppoUndelivered models.PpoUndelivered
+	err := json.Unmarshal(data, &ppoUndelivered)
+	if err != nil {
+		return nil, err
+	}
+	return ppoUndelivered, nil
+}
+
+func convertPpoUndeliveredToRmMessage(message models.PubSubMessage) (*models.RmMessage, error) {
```

Comment:
this is passing by value, not by pointer (i.e. reference) which would be bad practice in C/C++... but I would probably want to defer to the expertise of @eldeal and @davidmort as to whether this is bad practice in Go

Reply:
@NickGrantONS I believe becuase `models.PubSubMessage` is an interface type this is the accepted way of handling it, the actual value being passed is a table of 2 pointers, one to the interface and another to the actual data.

Reply:
table? I will need to read the Go docs some more, but what you say kinda makes sense (it's just my lack of knowledge that's the problem here)

Reply:
Table probably isn't the correct term, just what I'd picked up reading this https://medium.com/@saiyerram/go-interfaces-pointers-4d1d98d5c9c6